tools/dep_updaters/update-zlib.sh

```diff
@@ -1,23 +1,46 @@
 #!/bin/sh
 set -e
-# Shell script to update zlib in the source tree to a specific version
+# Shell script to update zlib in the source tree to the most recent version.
+# Zlib rarely creates tags or releases, so we use the latest commit on the main branch.
+# See: https://github.com/nodejs/node/pull/47417
 
 BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
 DEPS_DIR="$BASE_DIR/deps"
 
-CURRENT_VERSION=$(grep "#define ZLIB_VERSION" "$DEPS_DIR/zlib/zlib.h" | sed -n "s/^.*VERSION \"\(.*\)\"/\1/p")
- 
-NEW_VERSION_ZLIB_H=$(curl -s "https://chromium.googlesource.com/chromium/src/+/refs/heads/main/third_party/zlib/zlib.h?format=TEXT" | base64 --decode)
+NEW_UPSTREAM_SHA1=$(git ls-remote "https://chromium.googlesource.com/chromium/src/third_party/zlib.git" HEAD | awk '{print $1}')
+NEW_VERSION=$(echo "$NEW_UPSTREAM_SHA1" | head -c 7)
 
-NEW_VERSION=$(printf '%s' "$NEW_VERSION_ZLIB_H" | grep "#define ZLIB_VERSION" | sed -n "s/^.*VERSION \"\(.*\)\"/\1/p")
+echo "Comparing $NEW_VERSION with current revision"
 
-echo "Comparing $NEW_VERSION with $CURRENT_VERSION"
+git remote add zlib-upstream https://chromium.googlesource.com/chromium/src/third_party/zlib.git
+git fetch zlib-upstream "$NEW_UPSTREAM_SHA1"
+git remote remove zlib-upstream
 
-if [ "$NEW_VERSION" = "$CURRENT_VERSION" ]; then
+# We exclude win32, GN-scraper.py and zlib.gyp from checking diff because they are not part of the zlib source
+DIFF_TREE=$(
+  git diff HEAD:deps/zlib "$NEW_UPSTREAM_SHA1" -- ':!win32' ':!GN-scraper.py' ':!zlib.gyp'
```

Comment:
We should use `--exit-code` here if we don't care about the actual diff.
```suggestion
  git diff --exit-code HEAD:deps/zlib "$NEW_UPSTREAM_SHA1" -- ':!win32' ':!GN-scraper.py' ':!zlib.gyp' || echo 1
```

And maybe we can skip the variable altogether and use this command as the `if` condition.

Reply:
I noticed that the filter `!` to exclude files doesn't work with deleted files, and `win32`, `GN-scraper.py`, `zlib.gyp` are considered as deleted, so it never skip the update routine.
So we have to find a way to exclude those files that are not part of the Chromium source. We can use the filter `--diff-filter=d`, but of course we exclude checking files that are deleted on the source repo. It doesn't seem to be a real issue, cause even if the latest commit only delete files, they will be deleted on the first commit that made some changes.